More intuitive environment variable handling

Zethson opened the pull request:
Hi,

fixes #170 .

```
export FLASK_DEBUG=0                                                                                                           
export MEGAQC_PRODUCTION=1                                                                                                  

>>> from environs import Env
>>> env = Env()
>>> env.read_env()
>>> print(env.bool("FLASK_DEBUG", False))
False
>>> print(env.bool("MEGAQC_PRODUCTION", False))
True
```

Signed-off-by: Zethson <lukas.heumos@posteo.net>

TMiguelT commented on megaqc/cli.py:
```diff
@@ -38,10 +42,14 @@ def cli():
         print(" * Environment variable FLASK_DEBUG is true - running in dev mode")
         os.environ["FLASK_ENV"] = "dev"
-    elif not os.environ.get("MEGAQC_PRODUCTION", False):
+    elif env.bool("MEGAQC_PRODUCTION", False):
```
Missing the `not`

TMiguelT reviewed (changes requested):
You will also need to look into `environs`-ifying other uses of `os.environ`, for consistent behaviour:
* `megaqc/wsgi.py`
* `megaqc/settings.py`
* `megaqc/utils/settings.py`

Also, the docs will need to explain somewhere about the `.env` file.

Zethson commented:
@TMiguelT Thanks for the review.

I've added consistent env reading to the other files and hope that I didn't miss any.
Tested it briefly and it seems to do what it is supposed to do.

Honestly don't think that we need the `.env` file reading here and therefore I removed it.

TMiguelT commented on megaqc/settings.py:
```diff
@@ -109,11 +112,11 @@ class ProdConfig(Config):
     else:
         SQLALCHEMY_HOST = "{}:{}".format(
-            os.environ.get("DB_HOST", "localhost"), os.environ.get("DB_PORT", "5432")
+            env.str("DB_HOST", "localhost"), env.str("DB_PORT", "5432")
```
Should the port be an `int` in order to get a better error message?
